language/move-lang/src/shared/mod.rs

```diff
@@ -197,6 +199,55 @@ pub fn shortest_cycle<'a, T: Ord + Hash>(
     path
 }
 
+//**************************************************************************************************
+// Compilation Env
+//**************************************************************************************************
+
+pub struct CompilationEnv {
+    flags: Flags,
+    errors: Errors,
+    // TODO(tzakian): Remove the global counter and use this counter instead
+    // pub counter: u64,
+}
+
+impl CompilationEnv {
+    pub fn from_flags(flags: Flags) -> Self {
+        Self {
+            flags,
+            errors: Vec::new(),
+        }
+    }
+
+    pub fn error(&mut self, e: Vec<(Loc, impl Into<String>)>) {
+        self.errors
+            .push(e.into_iter().map(|(loc, msg)| (loc, msg.into())).collect())
+    }
+
+    pub fn errors(&mut self, es: Errors) {
```

Comment:
I imagine you carried this name from various context functions. But I kind of hate it .... 

Maybe `extend_errors` is better. Its kind of a mess though with "getters"

```
pub fn error(mut self, , e: Vec<(Loc, impl Into<String>)>) // records an error
pub fn extend_errors(&mut self, es: Errors) // records many errors
pub fn errors(&self) -> &Errors // gets errors
pub fn errors_mut(&mut self) -> &mut Errors // gets mutable errors, not sure why this needs to exist though and really can't in the future when this is parallel 
```
The dissimilarity between `error` and `errors` is gross though ... Open to suggestions